synapse/storage/databases/main/media_repository.py

```diff
@@ -149,6 +150,11 @@ def __init__(
             self._drop_media_index_without_method,
         )
 
+        if hs.config.media.can_load_media_repo:
+            self.unused_expiration_time = hs.config.media.unused_expiration_time
+        else:
+            self.unused_expiration_time = None
```

Comment:
(I tested it this time. 😄 )

```suggestion
        if hs.config.media.can_load_media_repo:
            self.unused_expiration_time: Optional[
                int
            ] = hs.config.media.unused_expiration_time
        else:
            self.unused_expiration_time = None
```